libs/auth/src/java_card.ts

```diff
@@ -323,24 +323,25 @@ export class JavaCard implements Card {
   }
 
   /**
-   * We wrap this.readUser() because it:
+   * We wrap readCardDetails to create safeReadCardDetails because readCardDetails:
```

Comment:
Yeah great questions!

> Are there places where we want readCardDetails to throw?

We ultimately never want the top-level card reading operation to throw. A bad card shouldn't be able to make a machine crash. It should instead surface a screen like this when the card is inserted to login:

![image](https://user-images.githubusercontent.com/12616928/226632725-0784676f-9509-4d18-8259-695ac9c99274.png)

This screen is surfaced by the auth layer when the card layer indicates that a card is in the card reader but a user could not be extracted from the card: https://github.com/votingworks/vxsuite/blob/b5328da61d490fd54899882c8d1fd22f3337e5cb/libs/auth/src/dipped_smart_card_auth.ts#L511-L513

Helpers are of course a different story and can throw all they need to communicate up to the top-level operation that the card is somehow invalid. Which leads to your second question:

> Can we simplify by getting rid of the wrapper and making readCardDetails itself safe?

We _could_ make the lower-level `readCardDetails` safe, but that would almost certainly complicate things rather than simplify. The simplest way to make it safe would be to wrap everything in a try-catch (basically what we do today but without the wrapper function). The more nuanced way to make it safe would be to have both it and all downstream helpers return `Result` objects (which as I'm sure you've now seen allow packaging both success results and error results without throwing). The main challenge here would be a constant need to check `Result` objects, through layers and layers of helpers. I touch on this in this old thread: https://github.com/votingworks/vxsuite/pull/3037#discussion_r1119011184